### PR TITLE
feat: add the ability for users to add resources to their collections

### PR DESCRIPTION
**Pull request by chosww**
Adds Livewire support to Hearth, which is used to build a Livewire component for selecting resources users want to add to their resource collections. Also, updates ResourceCollection controller and requests to have ability to include resources on resource collection create, and update. 

**Testing**

- Verify that ResourceSelect component loads all available resources
- Verify that ResourceSelect component loads any resources those are already included in a collection
- Verify adding or removing resources from ResourceSelect

### CODE REVIEW

**Comment by greatislander** [comment]
@chosww Can you check the two failing tests and see if you can resolve them?

**Comment by chosww** [comment]
> @chosww Can you check the two failing tests and see if you can resolve them?

Yep, I am working on them -- the issue seems like caused by Hearth not able to find ResourceSelect component. I will let you know once I figure it out.

**Comment by chosww** [comment]
Hey @greatislander, I fixed the previous CI tests, but now I am getting code coverage failure because of the line I added in HearthCommand. Is HearthCommand something we should test or do I have to update ignore coverage check somewhere?

**Review by greatislander — CHANGES_REQUESTED** [review]

File: stubs/resources/views/livewire/resource-select.blade.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<div class="flow">
+    <div x-data="{message: @entangle('message')}" aria-live="assertive">
+        <p x-text="message"></p>
+    </div>
```
Comment (greatislander) [inline stubs/resources/views/livewire/resource-select.blade.php:4]: We can do this without needing to use Alpine at all:

```suggestion
    <div aria-live="assertive">
        @if($message)
        <p class="visually-hidden">
            {{ $message }}
        </p>
        @endif
    </div>
```

When the message is set to an empty string, Livewire will remove the visually-hidden paragraph. When the message is populated, Livewire will add it the div. Tested in VoiceOver and it works as expected!

File: stubs/resources/views/layouts/app.blade.php [hunk 2]
```diff
@@ -23,5 +23,7 @@
                 </div>
             </article>
         </main>
+        @livewireScripts
+        @vite('resources/js/app.js')
```
Comment (greatislander) [inline stubs/resources/views/layouts/app.blade.php:27]: I think this change should be made in `layouts/guest.blade.php` as well.

File: stubs/app/Http/Livewire/ResourceSelect.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    public string $message = '';
+
+
+    public function mount(?int $resourceCollectionId)
+    {
+        $this->availableResources = new Collection();
+        $this->selectedResources = new Collection();
+        if ($resourceCollectionId != null) {
+            $resourcesInCollection = ResourceCollection::find($resourceCollectionId)->resources()->get();
+            $this->availableResources = Resource::all()->diff($resourcesInCollection);
```
Comment (greatislander) [inline stubs/app/Http/Livewire/ResourceSelect.php:23]: In the [integration test](https://github.com/fluid-project/hearth/runs/8040218647?check_suite_focus=true#step:11:74) I see this error:

```
Error: Called 'diff' on Laravel collection, but could have been retrieved as a query.
```

A quick search found this: https://laracasts.com/discuss/channels/laravel/called-diff-on-laravel-collection-but-could-have-been-retrieved-as-a-query

As I understand it, the issue here is that it's much less efficient to query ALL resources, then remove some, when one could query resources with the specified IDs instead. So I think you should make the following change:

```suggestion
            $this->availableResources = Resource::whereNotIn($resourcesInCollection->pluck('id'))->get();
```

File: stubs/app/Http/Livewire/ResourceSelect.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<?php
```
Comment (greatislander) [inline stubs/app/Http/Livewire/ResourceSelect.php:1]: You need to run `composer format` as there are some fixable code style issues with this file.

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Pull request merged**